src/components/form/index.tsx

```diff
@@ -0,0 +1,148 @@
+import PopUp from '#components/pop-up'
+import styled from '@emotion/styled'
+import React from 'react'
+
+function Feedback() {
+  const [isDisplayed, setDisplay] = React.useState(false)
+
+  function handleSubmit(e: React.FormEvent<HTMLFormElement>): void {
```

Comment:
why not use `useCallback`?

Reply:
instead of useState? if so why?

Reply:
instead of function 
```ts
const handleSubmit = React.useCallback<React.FormEventHandler<HTMLFormElement>>(e => {
 ...
},  []}
```

Reply:
interesting, whats the benefits?

Reply:
you won't register `handleSubmit` every render. it's very common practice.
You can use functions and lambdas when you really know what you're doing, but since this is OSS and we have many levels of competence, it's better to just memoize all callbacks instead of confusing people

Reply:
This is premature optimization and not necessary. Do not wrap every inline function in `useCallback`
https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
> 
> 
> This is premature optimization and not necessary. Do not wrap every inline function in `useCallback`
> https://kentcdodds.com/blog/usememo-and-usecallback

i was just think about that article

Reply:
I'm not sure I agree with this. Wouldn't using `useCallback` everywhere be more confusing and obscure the meaning of it? I would use it  when you have performance issues which it can solve.

Reply:
> I'm not sure I agree with this. Wouldn't using `useCallback` everywhere be more confusing and obscure the meaning of it? I would use it when you have performance issues which it can solve.

100% this is a great resource making the same point.
https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
I am aware of this, but i'd rather it's used as a standard and not missed when needed, then the other way around

I can't expect all devs to be able to instinctively know when memoization can help pref and when it doesn't have effect, like I said before, you can use functions and lambdas if you know what you're doing, but it will make other devs confused why here we should and there we should do X or Y, consistency is important not for those who can, but for those who can't

Reply:
Nope. It's a preoptimization pattern, definitely not going to preoptimize due to assumed ignorance, if someone sees a use case for it or performance becomes an issue for something then it can be applied. 

Memoizing everything without a good reason can lead to the very performance issues it is meant to stop. As mentioned in the article https://kentcdodds.com/blog/usememo-and-usecallback